src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed partial class EditorConfig
+    {
+        /// <summary>
+        /// Takes a <see cref="Section.Name"/> and compiles the file glob
+        /// inside to a regex which recognizes the given language.
+        /// </summary>
+        public static string CompileSectionNameToRegEx(string sectionName)
+        {
+            // An editorconfig section name is a language for recognizing file paths
+            // defined by the following grammar:
+            //
+            // <path> ::= <path-list>
+            // <path-list> ::= <path-item> | <path-item> <path-list>
+            // <path-item> ::= "*"  | "**" | "?" | <char> | <choice> | <range>
+            // <char> ::= any unicode character
+            //
+            // PROTOTYPE(editorconfig): Below remains unimplemented
+            //
+            // <choice> ::= "{" <choice-list> "}" 
+            // <choice-list> ::= <path-list> | <path-list> "," <choice-list>
+            // <range> ::= "{" <integer> ".." <integer> "}"
+            // <integer> ::= "-" <digit-list> | <digit-list>
+            // <digit-list> ::= <digit> | <digit> <digit-list>
+            // <digit> ::= 0-9
+
+            var sb = new StringBuilder();
+            sb.Append('^');
+            var lexer = new SectionNameLexer(sectionName);
+            while (!lexer.IsDone)
+            {
+                var tokenKind = lexer.Lex();
+                switch (tokenKind)
+                {
+                    case TokenKind.BadToken:
+                        // Parsing failure
+                        return null;
```

Comment:
Consider using a Try pattern for this method if we have an explicit non-throwing failure case.

Reply:
By "Try pattern" do you mean prefixing the method name with Try, or also using a bool return and an out var? Since `null` is never a valid value, seems like the bool/out pattern is overkill in this situation.

Reply:
Try with null return si fine. 

---
In reply to: [191528129](https://github.com/dotnet/roslyn/pull/27119#discussion_r191528129) [](ancestors = 191528129)